### ISSUE TITLE
Scalar variables are not well supported

### ISSUE THREAD

**Issue by ggalibert** [nctoolbox/nctoolbox#84]
when accessing data from a scalar variable nctoolbox says:
```
Warning: An error occurred while reading "LATITUDE" in
/home/ggalibert/Documents/ANMN/PH100-1106_2.4/IMOS_ANMN-NSW_BCDKOSTUZ_20110620T124547Z_PH100_FV01_PH100-1106-WQM-15.5_END-20110721T230140Z_C-20150225T011059Z.nc. Cause:
Error using ncdataset/readdata (line 462)
Java exception occurred:
java.lang.IllegalArgumentException: java.lang.IllegalArgumentException
	at ucar.ma2.Array.copyToNDJavaArray(Array.java:770)
Caused by: java.lang.IllegalArgumentException
	at java.lang.reflect.Array.multiNewArray(Native Method)
	at java.lang.reflect.Array.newInstance(Unknown Source)
	at ucar.ma2.Array.copyToNDJavaArray(Array.java:768)

Error in ncdataset/data (line 159)
                    d = obj.readdata(variable);

Error in ncvariable/alldata (line 323)
                data = obj.dataset.data(name);

Error in ncvariable/data (line 151)
                        d = alldata(obj,1);

Error in ncvariable/subsref (line 275)
                                sref = obj.data;
 
> In ncdataset>ncdataset.readdata at 464
  In ncdataset>ncdataset.data at 159
  In ncvariable>ncvariable.alldata at 323
  In ncvariable>ncvariable.data at 151
  In ncvariable>ncvariable.subsref at 275 
Warning: Variable "LATITUDE" has no netcdf dimension associated with it. Errors may result from non CF compliant files. 
> In ncvariable>ncvariable.subsref at 276
```
Instead of using line 462 of ncdataset.m: `d = array.copyToNDJavaArray(); % this fails if the variable has no java shape/no dimension was assigned` , you could use:

```
if v.getSize == 1 
    d = array.copyTo1DJavaArray; 
else 
    d = array.copyToNDJavaArray(); 
end
```

And you would probably don't need the trick in the catch statement.

Example file: http://thredds.aodn.org.au/thredds/catalog/IMOS/eMII/demos/ANMN/NRS/NRSKAI/catalog.html?dataset=eMII/demos/ANMN/NRS/NRSKAI/IMOS_ANMN-NRS_ACESTZ_20140506T051319Z_NRSKAI_FV01_NRSKAI-1405-NXIC-CTD-104.49_END-20141104T031329Z_C-20150903T024338Z.nc

**Comment by rsignell-usgs** [comment]
@ggalibert , thanks so much for reporting this bug and for suggesting a fix.  If you find more issues, please let us know! 
@hohonuuli, okay to implement this?

**Comment by hohonuuli** [comment]
@ggalibert You get a gold star for not only reporting the issue but for providing a fix ... :star2: . Much appreciated!
I've made the changes you've suggested and am running tests on it right now.

**Comment by hohonuuli** [comment]
Just a note (for myself) that the above URL throws does not support `Content-Length`. Use [http://thredds.aodn.org.au/thredds/dodsC/IMOS/eMII/demos/ANMN/NRS/NRSKAI/IMOS_ANMN-NRS_ACESTZ_20140506T051319Z_NRSKAI_FV01_NRSKAI-1405-NXIC-CTD-104.49_END-20141104T031329Z_C-20150903T024338Z.nc](http://thredds.aodn.org.au/thredds/dodsC/IMOS/eMII/demos/ANMN/NRS/NRSKAI/IMOS_ANMN-NRS_ACESTZ_20140506T051319Z_NRSKAI_FV01_NRSKAI-1405-NXIC-CTD-104.49_END-20141104T031329Z_C-20150903T024338Z.nc) instead.

**Comment by hohonuuli** [comment]
Fix published in release 1.1.2 (off of master branch)

**Comment by ggalibert** [comment]
Cool! Thanks, the fix was actually suggested to me by @sspagnol :-)
